### ISSUE TITLE
[Bug] After PlexPy Update, PlexPy does not relaunch

### ISSUE THREAD

**Issue by majora2007** [Tautulli/Tautulli#562]
I have been experiencing this bug for the past few updates.

Steps to reproduce:
1. PlexPy notification of update
2. Perform update
3. PlexPy auto relauches
4. Attempt refresh of PlexPy browser tab

Expected:
1. After PlexPy update, PlexPy would reopen (relaunch)

Actual:
1. PlexPy does not relauch/reopen until user runs "python PlexPy.py"

Please look into this issue. Thanks.

Branch: master
Commit: 97c414d1ade6c4dd254104b801a4e1f8e0c41c93
Relevant Settings: N/A - Default settings
Python: 2.7 Windows 10
Log: [Here](http://pastebin.com/3PbwVQwg)

**Comment by JonnyWong16** [comment]
Make sure you provide the following information:
- [x] Branch
- [x] Version/Commit hash
- [x] Your operating system and python version
- [x] What you did?
- [x] What happened?
- [x] What you expected?
- [x] How can we reproduce your issue?
- [x] What are your (relevant) settings?
- [x] Include a link to your **FULL** (not just a few lines!) log file that has the error. Please use [Gist](http://gist.github.com) or [Pastebin](http://pastebin.com/).

**Comment by JonnyWong16** [comment]
From your logs, these are the correct paths?

```
Python path: E:\Python27\pythonw.exe
PlexPy path" E:\My Documents\git\plexpy\PlexPy.py
```

I just tested on Windows 10 Python 2.7.11. It restarts as expected.

**Comment by pax0707** [comment]
This started happening to me to recently on my Win7 PC.
Other PC with WIn 10 works fine.

```
22-vlj-2016 01:16:15 - INFO    :: MainThread : Checking to see if the database has all tables....
22-vlj-2016 01:16:15 - DEBUG   :: MainThread : Trying to execute: "git rev-parse HEAD" with shell in C:\Program Files\PlexPy
22-vlj-2016 01:16:16 - DEBUG   :: MainThread : Git output: eab6365af9f83baaa7a0b06a2b06afabb0f0a667
22-vlj-2016 01:16:16 - DEBUG   :: MainThread : Trying to execute: "git rev-parse --abbrev-ref HEAD" with shell in C:\Program Files\PlexPy
22-vlj-2016 01:16:16 - DEBUG   :: MainThread : Git output: dev
22-vlj-2016 01:16:16 - INFO    :: MainThread : Retrieving latest version information from GitHub
22-vlj-2016 01:16:16 - DEBUG   :: MainThread : Requesting URL via GET method: https://api.github.com/repos/drzoidberg33/plexpy/commits/dev
22-vlj-2016 01:16:18 - DEBUG   :: MainThread : Latest version is de9f60aa7ff7d56c7bb5b0aef762a8851245c010
22-vlj-2016 01:16:18 - INFO    :: MainThread : Comparing currently installed version with latest GitHub version
22-vlj-2016 01:16:18 - DEBUG   :: MainThread : Requesting URL via GET method: https://api.github.com/repos/drzoidberg33/plexpy/compare/de9f60aa7ff7d56c7bb5b0aef762a8851245c010...eab6365af9f83baaa7a0b06a2b06afabb0f0a667
22-vlj-2016 01:16:18 - DEBUG   :: MainThread : In total, 11 commits behind
22-vlj-2016 01:16:18 - INFO    :: MainThread : New version is available. You are 11 commits behind
22-vlj-2016 01:16:18 - INFO    :: MainThread : PlexPy PlexTV :: Requesting URLs for server...
22-vlj-2016 01:16:18 - INFO    :: MainThread : Writing configuration to file
22-vlj-2016 01:16:18 - INFO    :: MainThread : Writing configuration to file
22-vlj-2016 01:16:18 - INFO    :: MainThread : PlexPy Pmsconnect :: Requesting name from server...
22-vlj-2016 01:16:19 - INFO    :: MainThread : PlexPy PlexTV :: Requesting users list refresh...
22-vlj-2016 01:16:19 - INFO    :: MainThread : PlexPy PlexTV :: Users list refreshed.
22-vlj-2016 01:16:19 - INFO    :: MainThread : PlexPy Pmsconnect :: Requesting libraries list refresh...
22-vlj-2016 01:16:19 - INFO    :: MainThread : PlexPy Pmsconnect :: Libraries list refreshed.
22-vlj-2016 01:16:19 - INFO    :: MainThread : Starting PlexPy web server on http://0.0.0.0:8181/
22-vlj-2016 01:16:19 - INFO    :: MainThread : Web server authentication is enabled, username is 'pax0707'
22-vlj-2016 01:16:21 - INFO    :: MainThread : Scheduled background task: Check GitHub for updates
22-vlj-2016 01:16:21 - INFO    :: MainThread : Scheduled background task: Refresh Plex server URLs
22-vlj-2016 01:16:21 - INFO    :: MainThread : Scheduled background task: Refresh Plex server name
22-vlj-2016 01:16:21 - INFO    :: MainThread : Scheduled background task: Check for recently added items
22-vlj-2016 01:16:21 - INFO    :: MainThread : Scheduled background task: Check for Plex remote access
22-vlj-2016 01:16:21 - INFO    :: MainThread : Scheduled background task: Check for active sessions
22-vlj-2016 01:16:21 - INFO    :: MainThread : Scheduled background task: Refresh users list
22-vlj-2016 01:16:21 - INFO    :: MainThread : Scheduled background task: Refresh libraries list
22-vlj-2016 01:16:21 - INFO    :: MainThread : Scheduled background task: Backup PlexPy database
22-vlj-2016 01:16:32 - INFO    :: MainThread : Received signal: update
22-vlj-2016 01:16:41 - INFO    :: MainThread : Writing configuration to file
22-vlj-2016 01:16:41 - INFO    :: MainThread : PlexPy is updating...
22-vlj-2016 01:16:41 - DEBUG   :: MainThread : Trying to execute: "git pull origin dev" with shell in C:\Program Files\PlexPy
22-vlj-2016 01:16:50 - DEBUG   :: MainThread : Git output: From https://github.com/drzoidberg33/plexpy
 * branch            dev        -> FETCH_HEAD
   eab6365..de9f60a  dev        -> origin/dev
Updating eab6365..de9f60a
Fast-forward
 CHANGELOG.md                                       |  28 ++
 CONTRIBUTING.md                                    |  39 +-
 .../default/js/tables/notification_logs.js         |  99 +++++
 data/interfaces/default/logs.html                  |  65 ++-
 data/interfaces/default/settings.html              |  13 +-
 lib/openanything.py                                | 107 -----
 plexpy/__init__.py                                 |  55 ++-
 plexpy/datafactory.py                              |  88 ++++-
 plexpy/helpers.py                                  |  47 ++-
 plexpy/notification_handler.py                     | 436 +++++++++++----------
 plexpy/notifiers.py                                |  42 +-
 plexpy/version.py                                  |   2 +-
 plexpy/web_socket.py                               |   4 +-
 plexpy/webserve.py                                 |  11 +-
 14 files changed, 676 insertions(+), 360 deletions(-)
 create mode 100644 data/interfaces/default/js/tables/notification_logs.js
 delete mode 100644 lib/openanything.py
22-vlj-2016 01:16:50 - INFO    :: MainThread : PlexPy is restarting...
22-vlj-2016 01:16:50 - INFO    :: MainThread : Restarting PlexPy with ['C:\\Program Files\\Python\\pythonw.exe', 'C:\\Program Files\\PlexPy\\PlexPy.py', '--nolaunch']
22-vlj-2016 01:22:11 - INFO    :: MainThread : Checking to see if the database has all tables....
22-vlj-2016 01:22:11 - DEBUG   :: MainThread : Altering database. Updating database table notify_log.
22-vlj-2016 01:22:11 - DEBUG   :: MainThread : Altering database. Updating database table notify_log.
22-vlj-2016 01:22:11 - DEBUG   :: MainThread : Trying to execute: "git rev-parse HEAD" with shell in C:\Program Files\PlexPy
22-vlj-2016 01:22:11 - DEBUG   :: MainThread : Git output: de9f60aa7ff7d56c7bb5b0aef762a8851245c010
22-vlj-2016 01:22:11 - DEBUG   :: MainThread : Trying to execute: "git rev-parse --abbrev-ref HEAD" with shell in C:\Program Files\PlexPy
22-vlj-2016 01:22:11 - DEBUG   :: MainThread : Git output: dev
22-vlj-2016 01:22:11 - INFO    :: MainThread : Retrieving latest version information from GitHub
22-vlj-2016 01:22:11 - DEBUG   :: MainThread : Requesting URL via GET method: https://api.github.com/repos/drzoidberg33/plexpy/commits/dev
22-vlj-2016 01:22:13 - DEBUG   :: MainThread : Latest version is d5f4a1a48a70a9804373ed98930447e0ea124a5c
22-vlj-2016 01:22:13 - INFO    :: MainThread : Comparing currently installed version with latest GitHub version
22-vlj-2016 01:22:13 - DEBUG   :: MainThread : Requesting URL via GET method: https://api.github.com/repos/drzoidberg33/plexpy/compare/d5f4a1a48a70a9804373ed98930447e0ea124a5c...de9f60aa7ff7d56c7bb5b0aef762a8851245c010
22-vlj-2016 01:22:13 - DEBUG   :: MainThread : In total, 1 commits behind
22-vlj-2016 01:22:13 - INFO    :: MainThread : New version is available. You are 1 commits behind
22-vlj-2016 01:22:13 - INFO    :: MainThread : PlexPy PlexTV :: Requesting URLs for server...
22-vlj-2016 01:22:13 - INFO    :: MainThread : Writing configuration to file
22-vlj-2016 01:22:14 - INFO    :: MainThread : Writing configuration to file
22-vlj-2016 01:22:14 - INFO    :: MainThread : PlexPy Pmsconnect :: Requesting name from server...
22-vlj-2016 01:22:14 - INFO    :: MainThread : PlexPy PlexTV :: Requesting users list refresh...
22-vlj-2016 01:22:15 - INFO    :: MainThread : PlexPy PlexTV :: Users list refreshed.
22-vlj-2016 01:22:15 - INFO    :: MainThread : PlexPy Pmsconnect :: Requesting libraries list refresh...
22-vlj-2016 01:22:15 - INFO    :: MainThread : PlexPy Pmsconnect :: Libraries list refreshed.
22-vlj-2016 01:22:15 - INFO    :: MainThread : Starting PlexPy web server on http://0.0.0.0:8181/
22-vlj-2016 01:22:15 - INFO    :: MainThread : Web server authentication is enabled, username is 'pax0707'
22-vlj-2016 01:22:16 - INFO    :: MainThread : Scheduled background task: Check GitHub for updates
22-vlj-2016 01:22:16 - INFO    :: MainThread : Scheduled background task: Refresh Plex server URLs
22-vlj-2016 01:22:16 - INFO    :: MainThread : Scheduled background task: Refresh Plex server name
22-vlj-2016 01:22:16 - INFO    :: MainThread : Scheduled background task: Check for recently added items
22-vlj-2016 01:22:16 - INFO    :: MainThread : Scheduled background task: Check for Plex remote access
22-vlj-2016 01:22:16 - INFO    :: MainThread : Scheduled background task: Check for active sessions
22-vlj-2016 01:22:16 - INFO    :: MainThread : Scheduled background task: Refresh users list
22-vlj-2016 01:22:16 - INFO    :: MainThread : Scheduled background task: Refresh libraries list
22-vlj-2016 01:22:16 - INFO    :: MainThread : Scheduled background task: Backup PlexPy database
22-vlj-2016 01:23:21 - DEBUG   :: Thread-15 : PlexPy Helpers :: Image uploaded to Imgur.
22-vlj-2016 01:23:22 - INFO    :: Thread-15 : PlexPy Notifiers :: Email notification sent.
22-vlj-2016 01:24:22 - INFO    :: MainThread : Received signal: update
22-vlj-2016 01:24:31 - INFO    :: MainThread : Writing configuration to file
22-vlj-2016 01:24:31 - INFO    :: MainThread : PlexPy is updating...
22-vlj-2016 01:24:31 - DEBUG   :: MainThread : Trying to execute: "git pull origin dev" with shell in C:\Program Files\PlexPy
22-vlj-2016 01:24:35 - DEBUG   :: MainThread : Git output: From https://github.com/drzoidberg33/plexpy
 * branch            dev        -> FETCH_HEAD
   de9f60a..d5f4a1a  dev        -> origin/dev
Updating de9f60a..d5f4a1a
Fast-forward
 CONTRIBUTING.md                       | 6 ++++--
 README.md                             | 6 ++++--
 data/interfaces/default/settings.html | 2 +-
 3 files changed, 9 insertions(+), 5 deletions(-)
22-vlj-2016 01:24:35 - INFO    :: MainThread : PlexPy is restarting...
22-vlj-2016 01:24:35 - INFO    :: MainThread : Restarting PlexPy with ['C:\\Program Files\\Python\\pythonw.exe', 'C:\\Program Files\\PlexPy\\PlexPy.py', '--nolaunch']
```

**Comment by majora2007** [comment]
@JonnyWong16 Yes those paths are correct. Log looks perfectly fine, but no matter how many times I refresh the browser, it doesn't come up until I run "pythonw.exe PlexPy.py" manually.

I assume because pythonw is windows-specific, it handle's Window's \'s different than standard python.

**Comment by JonnyWong16** [comment]
Does this only happen if you run PlexPy with `pythonw PlexPy.py`? Are you using the shortcut in the startup folder method?

Try running with `python PlexPy.py`, then clicking the "Restart" button in the settings. Does it restart then?

**Comment by RobinMM** [comment]
hi there, i am having same issue on windows 8.1, in the beginning it worked, but now it never seems to launch after an update.

**Comment by Hellowlol** [comment]
@Nyghto Aswer the same question as JonnyWong16 asked majora2007.

**Comment by RobinMM** [comment]
Actually should have read better before posting, sorry for that. Restarting didn't work, i checked my folders and i had python installed in program files. 

Reinstalled python in C:\Python27, launched plexpy and tried restarting it, it works :)
Haven't been able to check if updating works, because plexpy is already up to date. Will post an update about this.

**Comment by majora2007** [comment]
Yes, I am using shortcut for startup folder and that always works. 

Just tried your steps Jonny and low and behold, the error shows. There is an issue with escaping spaces (or normalizing slashes) it looks like.

![image](https://cloud.githubusercontent.com/assets/735851/13218457/46ae6cae-d92f-11e5-89d6-8e9b5d493fe9.png)

**Comment by gchingon** [comment]
Have the sane issue on OS X. I usually run $git pull plexpy
Then restart my Mac so it load from daemon and not have to keep a terminal window open just for plexpy. 

Yes I know I could 
$screen python plexpy.

**Comment by JonnyWong16** [comment]
@majora2007 It looks like it could be due to python not installed to the default location (`C:\Python27`) as @Nyghto mentioned. Or it's because there is a space in the folder path (although mine is also in "My Documents" and it works fine).

**Comment by JonnyWong16** [comment]
@elgallo So does PlexPy not start up after you restart your computer?

**Comment by Hellowlol** [comment]
Just escape white space for windows.

**Comment by JonnyWong16** [comment]
But it works on my Windows 10 in "My Documents"...

**Comment by pax0707** [comment]
```
C:\Program Files\PlexPy>python PlexPy.py
23-vlj-2016 21:48:21 - INFO :: MainThread : Checking to see if the database has all tables....
23-vlj-2016 21:48:21 - DEBUG :: MainThread : Trying to execute: "git rev-parse HEAD" with shell in C:\Program Files\PlexPy
23-vlj-2016 21:48:22 - DEBUG :: MainThread : Git output: 9ef389d335e9077b6467172d0e049d8b8c8994c2
23-vlj-2016 21:48:22 - DEBUG :: MainThread : Trying to execute: "git rev-parse --abbrev-ref HEAD" with shell in C:\Program Files\PlexPy
23-vlj-2016 21:48:22 - DEBUG :: MainThread : Git output: dev
23-vlj-2016 21:48:22 - INFO :: MainThread : Retrieving latest version information from GitHub
23-vlj-2016 21:48:22 - DEBUG :: MainThread : Requesting URL via GET method: https://api.github.com/repos/drzoidberg33/plexpy/commits/dev
23-vlj-2016 21:48:23 - DEBUG :: MainThread : Latest version is 9ef389d335e9077b6467172d0e049d8b8c8994c2
23-vlj-2016 21:48:23 - INFO :: MainThread : PlexPy is up to date
23-vlj-2016 21:48:23 - INFO :: MainThread : PlexPy PlexTV :: Requesting URLs for server...
23-vlj-2016 21:48:23 - INFO :: MainThread : Writing configuration to file
23-vlj-2016 21:48:24 - INFO :: MainThread : Writing configuration to file
23-vlj-2016 21:48:24 - INFO :: MainThread : PlexPy Pmsconnect :: Requesting name from server...
23-vlj-2016 21:48:24 - INFO :: MainThread : PlexPy PlexTV :: Requesting users list refresh...
23-vlj-2016 21:48:25 - INFO :: MainThread : PlexPy PlexTV :: Users list refreshed.
23-vlj-2016 21:48:25 - INFO :: MainThread : PlexPy Pmsconnect :: Requesting libraries list refresh...
23-vlj-2016 21:48:25 - INFO :: MainThread : PlexPy Pmsconnect :: Libraries list refreshed.
23-vlj-2016 21:48:25 - INFO :: MainThread : Starting PlexPy web server on http://0.0.0.0:8181/
23-vlj-2016 21:48:25 - INFO :: MainThread : Web server authentication is enabled, username is 'pax0707'
23-vlj-2016 21:48:26 - INFO :: MainThread : Scheduled background task: Check GitHub for updates
23-vlj-2016 21:48:26 - INFO :: MainThread : Scheduled background task: Refresh Plex server URLs
23-vlj-2016 21:48:26 - INFO :: MainThread : Scheduled background task: Refresh Plex server name
23-vlj-2016 21:48:26 - INFO :: MainThread : Scheduled background task: Check for recently added items
23-vlj-2016 21:48:26 - INFO :: MainThread : Scheduled background task: Check for Plex remote access
23-vlj-2016 21:48:26 - INFO :: MainThread : Scheduled background task: Check for active sessions
23-vlj-2016 21:48:26 - INFO :: MainThread : Scheduled background task: Refresh users list
23-vlj-2016 21:48:26 - INFO :: MainThread : Scheduled background task: Refresh libraries list
23-vlj-2016 21:48:26 - INFO :: MainThread : Scheduled background task: Backup PlexPy database
23-vlj-2016 21:48:50 - INFO :: MainThread : Received signal: restart
23-vlj-2016 21:49:00 - INFO :: MainThread : Writing configuration to file
23-vlj-2016 21:49:00 - INFO :: MainThread : PlexPy is restarting...
23-vlj-2016 21:49:00 - INFO :: MainThread : Restarting PlexPy with ['C:\\Program Files\\Python\\python.exe', 'C:\\Program Files\\PlexPy\\PlexPy.py', '
--nolaunch']

C:\Program Files\PlexPy>C:\Program: can't open file 'Files\Python\python.exe': [Errno 2] No such file or directory
```

OS Name:                   Microsoft Windows 7 Enterprise
OS Version:                6.1.7601 Service Pack 1 Build 7601

It was working fine till some time ago.
Kinda ignored it and started it manually after updates.

**Comment by JonnyWong16** [comment]
This looks like a problem with spaces in the file path. I'll look into it.

**Comment by pax0707** [comment]
Working now.

**Comment by JonnyWong16** [comment]
Should be fixed in dev.

**Comment by JonnyWong16** [comment]
Fixed in v1.3.10.
